great_expectations_provider/operators/great_expectations.py

```diff
@@ -44,121 +78,404 @@ class GreatExpectationsOperator(BaseOperator):
+            },
+            {
+                "name": "update_data_docs",
+                "action": {"class_name": "UpdateDataDocsAction", "site_names": []},
+            },
+        ]
+
+        if (
+            os.getenv("AIRFLOW__LINEAGE__BACKEND")
+            == "openlineage.lineage_backend.OpenLineageBackend"
+            and self.use_open_lineage
+        ):
+            self.log.info(
+                "Found OpenLineage Connection, automatically connecting... "
+                "(This behavior may be turned off by setting use_open_lineage to False.)"
+            )
+            openlineage_host = os.getenv("OPENLINEAGE_URL")
+            openlineage_api_key = os.getenv("OPENLINEAGE_API_KEY")
+            openlineage_namespace = os.getenv("OPENLINEAGE_NAMESPACE")
+            if not (openlineage_host and openlineage_api_key and openlineage_namespace):
+                raise ValueError(
+                    "Could not find one of OpenLineage host, API Key, or Namespace environment variables."
+                    f"\nHost: {openlineage_host}\nAPI Key: *****\nNamespace: {openlineage_namespace}"
+                )
+            action_list.append(
+                {
+                    "name": "open_lineage",
+                    "action": {
+                        "class_name": "OpenLineageValidationAction",
+                        "module_name": "openlineage.common.provider.great_expectations",
+                        "openlineage_host": openlineage_host,
+                        "openlineage_apiKey": openlineage_api_key,
+                        "openlineage_namespace": openlineage_namespace,
+                        "job_name": f"validate_{self.task_id}",
+                    },
+                }
+            )
+        return action_list
+
+    def build_default_checkpoint_config(self):
+        """Builds a default checkpoint with default values."""
+        self.run_name = (
+            self.run_name
+            if self.run_name
+            else f"{self.task_id}_{datetime.now().strftime('%Y-%m-%d::%H:%M:%S')}"
+        )
+        checkpoint_config = CheckpointConfig(
+            name=self.checkpoint_name,
+            config_version=1.0,
+            template_name=None,
+            module_name="great_expectations.checkpoint",
+            class_name="Checkpoint",
+            run_name_template=self.run_name,
+            expectation_suite_name=self.expectation_suite_name,
+            batch_request=None,
+            action_list=self.build_default_action_list(),
+            evaluation_parameters={},
+            runtime_configuration={},
+            validations=None,
+            profilers=[],
+            ge_cloud_id=None,
+            expectation_suite_ge_cloud_id=None,
+        ).to_json_dict()
+        filtered_config = {k: v for k, v in checkpoint_config.items() if v}
+
+        return filtered_config
+
+    def execute(self, context: "Context") -> Union[CheckpointResult, Dict[str, Any]]:
```

Comment:
Since `TYPE_CHECKING` is already imported, it should be possible to use `Context` without the quotes.